lib/snapper_error_check.rb

```
@@ -73,17 +69,46 @@ def javascript_errors(page, log_path, browser=nil)
     ensure
       file.close
       log.close
+      browser.close
+    end
 
-      if contained  #need to close if this was not a browser object passed in
-        browser.close
+    return errors_found
+  end
+
+  # check the return code of a particular site.
+  # return true if 200, false otherwise
+  def page_errors?(page, log_path="logs")
+    response_error = false
+
+    if !Dir.exists?(log_path)
+      Dir.mkdir(log_path)
+    end
+
+    puts("Looking for page errors on #{page}")
+    file = File.open("#{log_path}/page_errors.log", "a")
+    log = Logger.new(file)
+
+    begin
+      # use httparty
+      response = HTTParty.get(page)
+
+      if response.header.to_s.include? 'HTTPOK'
+        puts("Valid response. OK")
+      else
+        puts("Invalid response. #{response.header}")
+        response_error = true
       end
+    rescue => why
```

Why do we rescue all exceptions? Either let it fail without swallowing the exception (it will print out the unhandled exception by default) or handle something more specific if it consistently fails with an error that isn't a 'big deal'.